services/route_manager/src/main/java/com/futurewei/alcor/route/service/Impl/NeutronRouterServiceImpl.java

```diff
@@ -469,6 +468,197 @@ public ConnectedSubnetsWebResponse getConnectedSubnets(String projectId, String
+        }
+        // TODO: existRoutes replaced with map
+        // Tracking operation type for each routing rule
+        for (NewRoutesRequest newRouteRequest : routes) {
+            //RouteEntry route = null;
+            // TODO: 1. check new-des if it is valid
+            String newRouteDestination = newRouteRequest.getDestination();
+            String newRouteNexthop = newRouteRequest.getNexthop();
+            boolean isDestinationValid = verifyCidrBlock(newRouteDestination);
+            if (!isDestinationValid) {
+                throw new DestinationInvalid("destination is invalid : " + newRouteDestination);
+            }
+            String[] newRouteDes = newRouteDestination.split("\\/");
+            String newNetworkIP = newRouteDes[0];
+            String newBitmask = newRouteDes[1];
+            int newBitmaskInt = Integer.parseInt(newBitmask);
+            // TODO: 2. flag check if it is from vpc/Neutron(lower priority) or subnet.
+            RouteEntry existRoute = existRoutesMap.get(newNetworkIP);
+            // dont find: Create (both)
+            if (existRoute == null) {
+                InternalRoutingRule internalRoutingRule = null;
+                if (isNeutronOrVPCLevelRoutingRule) {
+                    internalRoutingRule = constructNewInternalRoutingRule(OperationType.CREATE, RoutingRuleType.DEFAULT, existRoute, newRouteRequest);
+                } else {
+                    internalRoutingRule = constructNewInternalRoutingRule(OperationType.CREATE, RoutingRuleType.STATIC, existRoute, newRouteRequest);
+                }
+                String Des = internalRoutingRule.getDestination();
+                String Nexthop = internalRoutingRule.getNextHopIp();
+
+                updateRoutes.add(internalRoutingRule);
+                hostRouteToSubnet.add(new HostRoute(){{setNexthop(Nexthop);setDestination(Des);}});
+
+            } else { // could find:
+                // compare which one cover the other one
+                if (isNeutronOrVPCLevelRoutingRule) { // VPC/Neutron
+                    String[] existDes = existRoute.getDestination().split("\\/");
+                    String existBitmask = existDes[1];
+                    int existBitmaskInt = Integer.parseInt(existBitmask);
+                    if (newBitmaskInt <= existBitmaskInt) { // new routing rule bitmask is smaller or equal than old one, drop it
+                        continue;
+                    } else { // new routing rule bitmask is larger than old one, Create new rule with low priority
+
+                        InternalRoutingRule internalRoutingRule = constructNewInternalRoutingRule(OperationType.CREATE, RoutingRuleType.DEFAULT, existRoute, newRouteRequest);
+
+                        updateRoutes.add(internalRoutingRule);
+                        hostRouteToSubnet.add(new HostRoute(){{setNexthop(internalRoutingRule.getNextHopIp());setDestination(internalRoutingRule.getDestination());}});
+
```

Comment:
default routing rules don't need update back to subnet's host route or routing table.

Reply:
okay